crates/rattler_menuinst/src/lib.rs

```diff
@@ -0,0 +1,150 @@
+    #[error("Menu config location is not a file: {0:?}")]
+    MenuConfigNotAFile(PathBuf),
+}
+
+// Install menu items from a given schema file
+pub fn install_menuitems(
+    file: &Path,
+    prefix: &Path,
+    base_prefix: &Path,
+    platform: Platform,
+    menu_mode: MenuMode,
+) -> Result<Vec<Tracker>, MenuInstError> {
+    let text = std::fs::read_to_string(file)?;
+    let menu_inst: MenuInstSchema = serde_json::from_str(&text)?;
+    let placeholders = BaseMenuItemPlaceholders::new(base_prefix, prefix, platform);
+
+    let mut trackers = Vec::new();
+    for item in menu_inst.menu_items {
+        if platform.is_linux() {
+            #[cfg(target_os = "linux")]
+            if let Some(linux_item) = item.platforms.linux {
+                let command = item.command.merge(linux_item.base);
+                let linux_tracker = linux::install_menu_item(
+                    &menu_inst.menu_name,
+                    prefix,
+                    linux_item.specific,
+                    command,
+                    &placeholders,
+                    menu_mode,
+                )?;
+                trackers.push(Tracker::Linux(linux_tracker));
+            }
+        } else if platform.is_osx() {
+            #[cfg(target_os = "macos")]
+            if let Some(macos_item) = item.platforms.osx {
+                let command = item.command.merge(macos_item.base);
+                let macos_tracker = macos::install_menu_item(
+                    prefix,
+                    macos_item.specific,
+                    command,
+                    &placeholders,
+                    menu_mode,
+                )?;
+                trackers.push(Tracker::MacOs(macos_tracker));
+            };
+        } else if platform.is_windows() {
+            #[cfg(target_os = "windows")]
+            if let Some(windows_item) = item.platforms.win {
+                let command = item.command.merge(windows_item.base);
+                let tracker = windows::install_menu_item(
+                    prefix,
+                    windows_item.specific,
+                    command,
+                    &placeholders,
+                    menu_mode,
+                )?;
+                trackers.push(Tracker::Windows(tracker));
+            }
+        }
+    }
+
+    Ok(trackers)
+}
+
+/// Remove menu items from a given schema file
+pub fn remove_menu_items(tracker: &Vec<Tracker>) -> Result<(), MenuInstError> {
+    for el in tracker {
+        #[allow(unused)]
+        match el {
+            Tracker::MacOs(tracker) => {
+                #[cfg(target_os = "macos")]
+                macos::remove_menu_item(tracker).unwrap();
+            }
+            Tracker::Linux(tracker) => {
+                #[cfg(target_os = "linux")]
+                linux::remove_menu_item(tracker).unwrap();
+            }
+            Tracker::Windows(tracker) => {
+                #[cfg(target_os = "windows")]
+                windows::remove_menu_item(tracker)?;
+            }
```

Comment:
I think you want to just return one error here instead of an unwrap

Reply:
Good point! I think there is one more unwrap in the windows specific code that I marked with a TODO.

Reply:
Let's align the return type with all the functions, and then also make this function return the removed paths. This is useful in pixi for logging.

Reply:
It's not just paths, though, it's also registry entries on Windows, for example, or entries in the terminal profile configuration file.

Might be more useful to just render the `tracker` information in pixi.